x-pack/plugin/rank-rrf/src/internalClusterTest/java/org/elasticsearch/xpack/rank/rrf/RRFRetrieverBuilderIT.java

```diff
@@ -743,6 +747,43 @@ public void extractToSearchSourceBuilder(SearchSourceBuilder searchSourceBuilder
         expectThrows(UnsupportedOperationException.class, () -> client().prepareSearch(INDEX).setSource(source).get());
     }
 
+    public void testRRFFiltersPropagatedToKnnQueryVectorBuilder() {
+        final int rankWindowSize = 100;
+        final int rankConstant = 10;
+        SearchSourceBuilder source = new SearchSourceBuilder();
+        // this will retriever all but 7 only due to top-level filter
+        StandardRetrieverBuilder standardRetriever = new StandardRetrieverBuilder(QueryBuilders.matchAllQuery());
+        // this would have retrieved 7 and 6, but due to parent level filter, will retriever 6 and 3 instead
+        KnnRetrieverBuilder knnRetriever = new KnnRetrieverBuilder(
+            "vector",
+            null,
+            new TestQueryVectorBuilderPlugin.TestQueryVectorBuilder(new float[] { 7 }),
```

Comment:
Don't you have to add the `TestQueryVectorBuilderPlugin` plugin? Or is it already added somewhere and I didn't see it?

Reply:
That was my assumption too, but I think that since this will be rewritten to a standard `query_vector` in the coordinator during request rewrite (before reaching out to the shards), and that `retriever` is eitherway transient for `source`, we don't actually need to serialize `TestQueryVectorBuilder` to be transferred over the wire.